core/src/main/java/org/eclipse/hono/config/AbstractConfig.java

```diff
@@ -186,25 +175,21 @@ public final TrustOptions getTrustOptions() {
         switch (format) {
         case PEM:
             LOG.debug("using certificates from file [{}] as trust anchor", this.trustStorePath);
-            trustOptions = new PemTrustOptions().addCertPath(this.trustStorePath);
-            break;
+            return new PemTrustOptions().addCertPath(this.trustStorePath);
         case PKCS12:
             LOG.debug("using certificates from PKCS12 key store [{}] as trust anchor", this.trustStorePath);
-            trustOptions = new PfxOptions()
+            return new PfxOptions()
                     .setPath(getTrustStorePath())
                     .setPassword(getTrustStorePassword());
-            break;
         case JKS:
             LOG.debug("using certificates from JKS key store [{}] as trust anchor", this.trustStorePath);
-            trustOptions = new JksOptions()
+            return new JksOptions()
```

Comment:
I do see value in creating the options once only. However, we then should also return the already created instance in subsequent invocations of this method. WDYT?